dash-renderer/src/components/error/CallbackGraph/CallbackGraphContainer.react.js

```diff
@@ -283,13 +292,23 @@ function CallbackGraph() {
         }
     }
 
+    const {graphLayout} = profile;
+    const cyLayout = graphLayout || {
+        name: 'dagre',
+        padding: 10,
+        spacingFactor: 0.8,
+        // after initial layout, just use this again on later draws
+        // but we'll also save the layout whenever users interact with it
+        ready: setPresetLayout,
+    }
```

Comment:
The first render uses Dagre - after that we just save the actual layout and regurgitate it when rendering again. That means you can close and reopen the graph, and it'll be faster the second time (only really noticeable on a huge graph like dash-docs, but there it's 4-5 seconds the first time, half a second after that), as well as saving any zoom / pan / moved nodes from the last time you looked at the graph.

A bit hacky, I just stash the layout with the profiling info in the redux store.

Dagre seems to do a good job for all the apps I've found to play with. On the biggest one, dash-docs, it does make a few silly errors - 5 or 6 out of the hundreds of nodes are wildly out of place, with an obvious way to fix them. I played around with Dagre's options a bit but could only make this worse, and anyway the vast majority of the graph looks good. Other more reasonable graphs look fine AFAICT. But I'd love it if other folks could try it out and let me know what you think!